audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class NoiseGate {
+  /**
+   * A Noise gate allows audio signals to pass only when the registered volume
+   * is above a specified threshold.
+   * @param  {BaseAudioContext} context the audio context
+   * @param  {Number} options.channels the number of input and output 
+   *                                   channels, a maximum of two
+   * @param  {Number} options.attack milliseconds for gate to fully close  
+   * @param  {Number} options.attack milliseconds for gate to fully open
+   * @param  {Number} options.bufferSize the size of an onaudioprocess window
+   * @param  {Number} options.threshold gain level beneath which sound is muted
+   * @param  {Number} options.alpha weight for envelope follower affecting
+   *                                the tradeoff between delay and smoothness
+   */
+  constructor(context, options) {
+    if (!(context instanceof BaseAudioContext)) 
+      throw context + ' is not a valid audio context.';
+    if (options == null) options = {};
+    this.context_ = context;
+    let channels = options.channels || 1;
+    let bufferSize = options.bufferSize || 0;
+    this.threshold = options.threshold || 0;
+    this.attack = options.attack || 0;
+    this.release = options.release || 0;
+    // Alpha controls a tradeoff between the smoothness of the 
+    // envelope and its delay, with a higher value giving more smoothness at 
+    // the expense of delay and vice versa. It should be set experimentally to
+    // control this tradeoff. 0.90 has proven an effective default in offline
+    // observations.
+    this.alpha_ = options.alpha || 0.90;
+    if (channels > 2)
+      throw 'The maximum supported channels is two.';
```

Comment:
This occurs after setting default parameters to allow for the possibility for channels to be null and the default value to take effect.